src/corehost/common/pal.windows.cpp

```diff
@@ -253,17 +251,26 @@ bool pal::get_own_executable_path(string_t* recv)
     return true;
 }
 
-pal::string_t pal::to_palstring(const std::string& str)
+pal::string_t pal::utf8_palstring(const std::string& str)
 {
-    return t_converter.from_bytes(str);
+    // No need of explicit null termination, so pass in the actual length.
+    pal::string_t retval;
+    size_t size = ::MultiByteToWideChar(CP_UTF8, 0, str.c_str(), str.size(), nullptr, 0);
+    retval.resize(size, '\0');
+    if (size == 0)
+    {
+        return retval;
+    }
+    ::MultiByteToWideChar(CP_UTF8, 0, str.c_str(), str.size(), &retval[0], retval.size());
```

Comment:
I am mostly following what the CLR does in such cases, I took your feedback because it is the right thing to do. However, I am leaving the ignoring the error code to the callers of this library like util function.

https://github.com/dotnet/coreclr/blob/775003a4c72f0acc37eab84628fcef541533ba4e/src/dlls/mscoree/unixinterface.cpp#L59